lib/oper1.g

```diff
@@ -360,6 +360,40 @@ BIND_GLOBAL( "InstallOtherMethod",
     end );
 
 
+#############################################################################
+##
+#F  InstallEarlyMethod( <opr>,<method> )
+##
+##  <#GAPDoc Label="InstallEarlyMethod">
+##  <ManSection>
+##  <Func Name="InstallEarlyMethod" Arg="opr,method"/>
+##
+##  <Description>
+##  installs a special "early" function method <A>method</A> for the
+##  operation <A>opr</A>. An early method is special in that it bypasses
+##  method dispatch, and is always the first method to be called when
+##  invoking the operation. This can be used to avoid method selection
+##  overhead for certain special cases.
+##  <P/>
+##  Unlike with regular methods, no checks are performed on the
+##  arguments. Hence any operation can have at most one such early
+##  method for each arity (i.e., one early method taking 1 argument, one
+##  early method taking 2 arguments, etc.).
+##  <P/>
+##  If an early method detects that it is not applicable, it can resume
+##  regular method dispatch by invoking <Ref Func="TryNextMethod"/>.
+##  <P/>
+##  For example, we install early methods for <Ref Oper="First"/> that
+##  deal with internal lists, for which computing their types (and hence
+##  method selection) can be very expensive.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+BIND_GLOBAL( "InstallEarlyMethod", INSTALL_EARLY_METHOD );
+#TODO; store location info somehow
```

Comment:
What does this mean?

Reply:
I think it is a reminder to implement the idea in https://github.com/gap-system/gap/pull/4557#issuecomment-859024184 sometime (not necessarily in this PR).

Reply:
Ah, I see. Thanks :)